tango/__main__.py

```diff
@@ -291,53 +304,53 @@ def info(config: TangoGlobalSettings):
 def _run(
     config: TangoGlobalSettings,
     experiment: str,
-    directory: Optional[Union[str, os.PathLike]] = None,
+    workspace_dir: Optional[Union[str, os.PathLike]] = None,
     overrides: Optional[str] = None,
     include_package: Optional[Sequence[str]] = None,
-):
-    from tango.common.util import import_module_and_submodules
+) -> Path:
     from tango.executor import Executor
-    from tango.step_cache import StepCache
+    from tango.local_workspace import LocalWorkspace
     from tango.step_graph import StepGraph
+    from tango.workspace_server import WorkspaceServer
 
     # Read params.
     params = Params.from_file(experiment, params_overrides=overrides or "")
 
     # Import included packages to find registered components.
     # NOTE: The Executor imports these as well because it's meant to be used
     # directly, but we also need to import here in case the user is using a
-    # custom Executor or StepCache.
+    # custom Executor, StepCache, or Workspace.
     include_package: List[str] = list(include_package or [])
     include_package += params.pop("include_package", [])
     include_package += config.include_package or []
     for package_name in include_package:
-        import_module_and_submodules(package_name)
+        import_extra_module(package_name)
 
     # Prepare directory.
-    if directory is None:
+    if workspace_dir is None:
         from tempfile import mkdtemp
 
-        directory = mkdtemp(prefix="tango-")
+        workspace_dir = mkdtemp(prefix="tango-")
         click.echo(
-            "Creating temporary directory for run: " + click.style(f"{directory}", fg="yellow")
+            "Creating temporary directory for run: " + click.style(f"{workspace_dir}", fg="yellow")
         )
-    directory = Path(directory)
-    directory.mkdir(parents=True, exist_ok=True)
+    workspace_dir = Path(workspace_dir)
+    workspace_dir.mkdir(parents=True, exist_ok=True)
+    workspace = LocalWorkspace(workspace_dir)
 
-    # Initialize step graph, cache, and executor.
+    # Initialize step graph, server, and executor.
     step_graph = StepGraph(params.pop("steps", keep_as_dict=True))
-    step_cache = StepCache.from_params(
-        params.pop("cache", default={}), dir=directory / "step_cache"
-    )
-    executor = Executor.from_params(
-        params.pop("executor", default={}),
-        dir=directory,
-        step_cache=step_cache,
+    server = WorkspaceServer.on_free_port(workspace)
+    server.serve_in_background()
```

Comment:
I think we should have the option to disable spinning up the server.

Reply:
8d8670a